src/console.rs

```diff
@@ -200,5 +205,31 @@ impl SimpleTextInput for Console {
             }
         }
     }
+
+    fn read_key_timeout(&self, timeout_millis: u64) -> Option<Result<InputKey, Status>> {
```

Comment:
I'm not sure this belongs in library code.  The set of events that are interesting to an application at a particular moment is typically specific to the application.  The function won't work for you, either, after you add serial support.